### DISCUSSION TITLE
Self hosted

### DISCUSSION THREAD

**Discussion by Guray00** [streetwriters/notesnook#852] [Ideas]
Are you planning to release a "only self hosted" version?

**Comment by anndix** [comment]
I would be interested as in self hosted version.

**Comment by Chiogros** [comment]
It is planned for october.
See their [roadmap](https://notesnook.com/roadmap/).

**Comment by julianbrennum** [comment]
One year later, but still no updates regarding this. Do we know if there is a new date for the selfhosted server?

**Comment by sinsky** [comment]
The roadmap has been updated.
Self hosting sync server was In progress [until October 2023](https://web.archive.org/web/20231031125510/https://notesnook.com/roadmap/), but the status was changed to Planned.

**Reply by thecodrr** [reply]
I explained this in [another thread](https://github.com/streetwriters/notesnook/issues/2066):

> Audit is definitely on the roadmap but not one of the more prioritized things. This is because we aren't yet at a place where an audit would be fruitful aside from as a marketing strategy. We are changing too many things too fast so any audit will become invalid as soon as it lands. **That is why, plans for audit have been postponed until we are at a relatively stable place i.e. post v3, and certainly not before self-hosting.**

Work on self hosting will start post-v3.

**Comment by aadrian** [comment]
Any news regarding self-hosting?

**Comment by mannp** [comment]
No longer a Pro customer, as the self hosted option appears to be going backwards.

Edit: Actually paused on GPS, rather than cancelled, as 2 years is too long to promote an option, with no updated timeline, from the original Oct 2022 one.

Hardly using the product but still paying pro is not the action of someone who has given up, but rather someone who has become rather tired of waiting.

**Comment by craftzneko** [comment]
Hi @thecodrr 

just discovered notesnook, no idea why I never saw it before I have been looking for a secure note app for what seems like years but my requirements are self hosted only so perhaps I had skipped over this. I love the design and I love the direction you are taking the project and I am hopeful for an open source self hosted sync server to go with the apps. I want to set a realistic expectation for myself, as v3 has been released and you mentioned that work on this would begin after that, how long do you expect it to take before it becomes available, 1 month, 2-3 months, 6 months, a year or longer? I  realize it's always hard to put dates on these things but even a rough indication would help and show people its still on track

Thanks and all the best

**Reply by thecodrr** [reply]
We cannot give any expected, approximate or rough timeline for this, unfortunately. All I can guarantee is that it is in progress; nothing more, nothing less. Some users have given up thinking we have no plans to officially support self hosting. This is not true. Self hosting will happen (and soon) but as with everything else, we cannot set a fixed timeline until we are ready.

**Reply by mannp** [reply]
> Some users have given up thinking we have no plans to officially support self hosting. This is not true. Self hosting will happen (and soon)

Your current android app, v3.09 timeline and website shows self hosted as '(0/4) Planned', yet you state 'will happen (and soon)' and ... 'All I can guarantee is that it is in progress; nothing more, nothing less.', so shouldn't your timeline at least reflect this new current situation? 

As a current customer who has paid for pro for some time, but who has become confused with the messaging around this proposed feature, so has paused their GPS subscription; I am disappointed that you have chosen to thumbs down me, instead of keeping your customers appraised in all channels, of the current situation after circa 2yrs.

**Reply by thecodrr** [reply]
> so shouldn't your timeline at least reflect this new current situation?

It should. I'll update the roadmap right away.

> who has become confused with the messaging around this 

I am not sure why there's any confusion. Self hosting will be officially supported. That's our messaging. We have never said at any time or place that self hosting is abandoned so what's the confusion?

> I am disappointed that you have chosen to thumbs down me

I thumbs down'd you for a different reason: we have always been 100% transparent with our development process and roadmap, and we have always tried to include our users and customers in that process. We don't make empty promises saying we'll add something but never add it. Whatever we said we'll add/fix we have added & fixed. We don't give out ETAs because that just creates bad faith when we can't meet those expectations. So when someone comes and says, "I am cancelling my subscription because I am tired of waiting", it only serves to demotivate and demoralize the whole team. I don't want to sound ungrateful or arrogant – your support is literally what's keeping us afloat but leaving comments like this helps no one.

> instead of keeping your customers appraised in all channels, of the current situation after circa 2yrs.

We have. Check this issue and the latest comments: https://github.com/streetwriters/notesnook-sync-server/issues/1.

**Reply by craftzneko** [reply]
Thanks for the update. I have only just discovered this project so for me the waiting starts now. I appreciate software development takes time from personal experience and it does look like great work has been achieved on v3. I am watching this project and given it a star I have high hopes for the future.

On a side note, do you have any guidance on how to compile from source for your client apps, in my case targeting arm. Not really done this before, if you have any pointers on where to start with eletron based apps please advise. I am assuming that if i DONT want to sync i can use the client apps and store data locally?

**Reply by thecodrr** [reply]
> do you have any guidance on how to compile from source for your client apps

All the steps are listed here: https://github.com/streetwriters/notesnook/tree/master/apps/desktop

**Reply by thecodrr** [reply]
You can also join the Discord community (or create another discussion here) if you face any issue.

**Reply by craftzneko** [reply]
Cheers i will join the discord as well :-)

That guide seems to build and then run but i was looking to compile a binary myself from source, am i missing something?

**Reply by mannp** [reply]
> I am not sure why there's any confusion. Self hosting will be officially supported. That's our messaging. We have never said at any time or place that self hosting is abandoned so what's the confusion?

I don't recall ever saying anything of the sort, perhaps you are mixing responses from others here, as my confusion comes from 'official channels' facts on the timeline against statements made in multiple other places, that do no correlate.

> I thumbs down'd you for a different reason: we have always been 100% transparent with our development process and roadmap, and we have always tried to include our users and customers in that process. We don't make empty promises saying we'll add something but never add it. Whatever we said we'll add/fix we have added & fixed. We don't give out ETAs because that just creates bad faith when we can't meet those expectations. So when someone comes and says, "I am cancelling my subscription because I am tired of waiting", it only serves to demotivate and demoralize the whole team. I don't want to sound ungrateful or arrogant – your support is literally what's keeping us afloat but leaving comments like this helps no one.

Again, I have not commented on any of the above at all, I commented in a thread regarding self hosting, that was originally timelined for Oct 2022 and for one reason or another is still ongoing.

Even with my limited use, I continued to pay pro and my post was to say, I can no longer justify to pay pro when the feature I am waiting for is only planned on the official timeline.

> We have. Check this issue and the latest comments: [streetwriters/notesnook-sync-server#1](https://github.com/streetwriters/notesnook-sync-server/issues/1).

This is referring to the official timeline in the android app and website not correlating with the multiple other posts on the matter.

> "I am cancelling my subscription because I am tired of waiting", it only serves to demotivate and demoralize the whole team.

It is a great app, but I'd prefer to have my data on my own infra and pay a reduced pro for using my own infra.

**Reply by thecodrr** [reply]
> This is referring to the official timeline in the android app and website not correlating with the multiple other posts on the matter.

I am confused. What other posts? I'd love a link so I can clarify.

**Reply by mannp** [reply]
> > This is referring to the official timeline in the android app and website not correlating with the multiple other posts on the matter.
> 
> I am confused. What other posts? I'd love a link so I can clarify.

This thread, 'will happen (_and soon_)'.

https://github.com/streetwriters/notesnook-sync-server/issues/1

`An update for those waiting: v3 release is done. Next step (after we enter a little bit more stable territory with v3) is self hosting as promised.`

Start self hosting or release? 3 weeks ago.

Official timeline - '(0/4) Planned'.

It is 'planned', 'coming soon', 'not started', not entirely sure.

Anyway.

**Reply by thecodrr** [reply]
Okay, so what's the confusion? Work on self hosting started when we open sourced the sync server. It's always been "in progress". That "planned" state (which will be updated) or "and soon" stage is related to its release and finalization. The only difference between now and then is now it is our primary focus, before it was only being worked on in the background.

I am observing a lot of bad faith from your end as if we are deliberately trying to miscommunicate and confuse users. I get that its frustrating to wait for something, but all this nitpicking over words is unnecessary and helps no one.

Our goal has been clear from day one: users will be able to self host for free, on their machines, unlimited, and unhindered. Nothing has changed in that regard.

The only question that's left is "when" but unfortunately, we cannot promise a date just now. It will happen when it will happen. Everyone's patience is appreciated.

**Reply by mannp** [reply]
> I am observing a lot of bad faith from your end as if we are deliberately trying to miscommunicate and confuse users.

There is really no need to get personal.

I am quoting your comments, made recently, not communicating anything myself.

No longer waiting, cancelled and moved to another solution.

**Reply by thecodrr** [reply]
> There is really no need to get personal.

That wasn't my intention. I just hope to make it clear that there is no deliberate attempt to misguide or miscommunicate. In fact, the whole reason we do not give out timelines, rough or approximate, is to avoid giving false hope to users.

> No longer waiting, cancelled and moved to another solution.

That is unfortunate but I wish you best of luck.

**Reply by rebelC0der** [reply]
Also, just found this app. Was positively surprised by the UI quality, performance and system resource usage.

But after playing around with it, I discovered a few negative and "classic" trends/issues, that "plague" a lot of good open-source notes taking apps, that made me and a lot of others to just skip them, and a lot of them just lost a huge user base because of those odd limitations.

Here are two super-odd limitations that stopped me from going forward with Notesnook (and 100 other Notes apps):

1. I see no clear way of selecting a directory, where notes/attachments will be stored. I was going to use my Nextcloud instance to sync my notes between devices, and not special Sync service is needed in this case.
2. When I try to add an attachment image to my note, I am asked/nagged to pay for a subscription/storage on the Notesnook server? Why can't I store stuff locally or in a cloud service of my choice?

Yes, this is a free and open-source software, but I was happy to donate money if it did not have these odd and silly limitations. Off to search for another notes app.

**Reply by thecodrr** [reply]
Good luck in finding another app that better suits your needs!

**Comment by Dvalin21** [comment]
Just discovered Notesnook. Been searching for a long time for a evernote alternative. Would love to see a self hosted, made turn key or even dockerized

**Comment by shuvashish76** [comment]
> https://github.com/streetwriters/notesnook-sync-server/issues/1#issuecomment-2316908876

@thecodrr I just tried android v3.0.15. There is no option to change instance. Is it planned for the next version then?

**Reply by ammarahm-ed** [reply]
It's there in Customization > Servers

![Screenshot_2024-08-30-10-59-29-05_293ad2e737335fce7c2b783e8aed96fe.jpg](https://github.com/user-attachments/assets/a39e66fa-5974-422a-9294-d92ea7eb9465)

**Reply by shuvashish76** [reply]
Thanks, Removed AF from F-Droid: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/15660

**Reply by serrq** [reply]
Sorry for my ignorance in this discipline. What is AF?

**Reply by ammarahm-ed** [reply]
> Sorry for my ignorance in this discipline. What is AF? 

AF stands for anti features.

**Reply by shuvashish76** [reply]
@serrq https://f-droid.org/en/docs/Anti-Features/

**Comment by poudenes** [comment]
Any news about the release date for a self hosted version (I hope docker)

**Comment by grivanov** [comment]
Hello, just to bump this up. :) There are people hoping this gets completed. As I see from the roadmap, only documentation remains to be written?

**Comment by ArthurK0** [comment]
Hello Notesnook team,

First, thank you for creating a solid privacy-focused note-taking platform — I’ve been using Notesnook extensively and appreciate the effort you’ve put into the product.

I’m writing to ask for some clarification regarding the self-hosting support, particularly the future direction and level of commitment from the team. From the current roadmap and discussions, it seems that self-hosting is still in progress but documentation and full stability are not yet delivered.

Since I rely on Notesnook for important data, it’s important for me to understand whether self-hosting is being actively developed toward full support, or whether it is being deprioritised in favour of the hosted service. If self-hosting is entering a passive/low-priority phase without clear timelines, that would directly influence my decision on whether to migrate to another solution.

Could you please clarify:
- Whether self-hosting remains a priority for the team.
- Whether we can expect an official stable release with documentation soon and a rough timeline.
- How you see the future of self-hosting in terms of ongoing maintenance and support.

Thank you for your time and clarification.

**Reply by netstat-peanut** [reply]
I appreciate your optimism (or rather, the optimism expressed by whichever LLM generated your message) but this is arguably a rhetorical question at this point. 

3 1/2 years after Notesnook "[went open source](https://blog.notesnook.com/notesnook-is-going-open-source/)" the only efforts to progress self-hosting have been [on the part of its users](https://github.com/streetwriters/notesnook-sync-server/issues/20). Developers stopped responding years ago. [Commercial options have evolved](https://web.archive.org/web/20250701000000*/https://notesnook.com/pricing/) to capitalize on interest garnered by a  perpetually-looming self-hosted destination with no mention of _when_ that might be expected. With that sweet, sweet taste of capitalism this has only progressed towards Camp Standardnotes.

The only (somehow more optimistic?) possibility is that former/proprietary systems/infrastructure were so starved of consistency and/or documentation that an otherwise-straightforward transition to self-hosting would expose mountains of duct tape and bailing wire; hence the delay. OK but almost 4 years of clean-up/refactoring? 

I too would have enjoyed a meaningful commitment to self-hosting but that won't be found with Notesnook.
